docs/topics/developer-tools.rst

```diff
@@ -0,0 +1,248 @@
+feature, which allows you to inspect the underlying HTML code of 
+any webpage. To demonstrate the Inspector, let's take a 
+look at the `quotes.toscrape.com`_-site.
+
+On the site we have a total of ten quotes from various authors with specific
+tags, as well as the Top Ten Tags. Let's say we want to extract all the quotes 
+on this page, without any meta-information about authors, tags, etc. 
+
+Instead of viewing the whole source code for the page, we can simply right click 
+on a quote and select ``Inspect Element (Q)``, which opens up the `Inspector`.
+In it you should see something like this:
+
+.. image:: _images/inspector_01.png
+   :width: 777
+   :height: 469
+   :alt: Firefox's Inspector-tool
+
+The interesting part for us is this:
+
+.. code-block:: html
+
+    <div class="quote" itemscope="" itemtype="http://schema.org/CreativeWork">
+      <span class="text" itemprop="text">(...)</span>
+      <span>(...)</span>
+      <div class="tags">(...)</div>
+    </div>
+
+If you hover over the first ``div`` directly above the ``span``-tag highlighted
+in the screenshot, you'll see that the corresponding section of the webpage gets
+highlighted as well. So now we have a section, but we can't find our quote text
+anywhere.
+
+The advantage of the `Inspector` is that it automatically expands and collapses
+sections and tags of a webpage, which greatly improves readability. You can
+expand and collapse a tag by clicking on the arrow in front of it or by double
+clicking directly on the tag. If we expand the ``span``-tag with the ``class=
+"text"`` we will see the quote-text we clicked on. The `Inspector` lets you
+copy XPaths to selected elements. Let's try it out: Right-click on the ``span``-
+tag, select ``Copy > XPath`` and paste it in the scrapy shell like so::
+
+    >>> scrapy shell "http://quotes.toscrape.com/"
+    (...)
+    >>> response.xpath('/html/body/div/div[2]/div[1]/div[1]/span[1]/text()').extract()
+    ['"The world as we have created it is a process of our thinking. It cannot be changed without changing our thinking.”]
+
+Adding ``text()`` at the end we are able to extract the first quote with this 
+basic selector. But this XPath is not really that clever. All it does is
+go down a desired path in the source code starting from ``html``. So let's 
+see if we can refine our XPath a bit: 
+
+If we check the `Inspector` again we'll see that directly beneath our 
+expanded ``div``-tag we have eight identical ``div``-tags, each with the 
+same attributes as our first. If we expand any of them, we'll see the same 
+structure as with our first quote: Two ``span``-tags and one ``div``-tag. We can
+expand each ``span``-tag with the ``class="text"`` inside our ``div``-tags and 
+see each quote. With this knowledge we can refine our XPath: Instead of a path
+to follow, we'll simply select all ``span``-tags with the ``class="text"``:: 
+
+    >>> response.xpath('//span[@class="text"]/text()').extract()
```

Comment:
Even though `'//span[@class="text"]/text()'` works for this website specifically, but maybe we should promote the usage of the new `has-class()` XPath function here:

```python
response.xpath('//span[has-class("text")]/text()').getall()
```

Comparing `@class="text"` can make your spider brittle, because if the website developer adds a new class to the quote elements (`<span class="text odd"... >`), the original selector wouldn't work anymore.